source/MaterialXRuntime/RtLook.cpp

```diff
@@ -53,6 +53,12 @@ RtAttribute RtLookGroup::getActiveLook() const
 
 void RtLookGroup::addLook(const RtObject& look)
 {
+    PvtPrim* pprim = PvtObject::ptr<PvtPrim>(look);
+    const string& typeName = pprim->getTypeInfo()->getShortTypeName();
+    if (typeName != RtLook::typeName() && typeName != RtLookGroup::typeName())
+    {
+        throw ExceptionRuntimeError("Cannot add invalid type to look group: '" + typeName + "'");
+    }
```

Comment:
I can do:
auto prim = lookgroup.asA<RtPrim>();
auto looks = prim.getRelationship("looks");
looks.addTarget(anything);
and bypass the check. This is why I think the RtPrim::createRelationship function should include a validation std::function parameter. That function can then be used in RtRelationship::addTarget( ) to validate at the lowest possible level that a RtPrim can be added as target.
I could then ask for that function in the node editor and use it to disallow invalid connections without having to use try/catch logic to validate if a connection can be done.

Reply:
Yes. This is true. I didn't try to address that level. I don't recall what if any resolution came of the chat with Niklas. Another approach could be we don't expose RtRelationships at all as this should be an implementation detail (we could have or could still change this). So there is a only public interfaces such as addLook() for lookgroups, addMaterialAssign() for looks etc.

Reply:
As per discussion, if we are to resolve this as a separate PR, do we want to remove these checks for now, or we can revise them when lower level checks are added.